tsconfig.json

```diff
@@ -21,7 +21,7 @@
     // "removeComments": true,                      /* Do not emit comments to output. */
     // "noEmit": true,                              /* Do not emit outputs. */
     // "importHelpers": true,                       /* Import emit helpers from 'tslib'. */
-    // "downlevelIteration": true,                  /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    "downlevelIteration": true,                  /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
```

Comment:
I'm curious. why we need this?

Reply:
on MemoryDatastore, we iterate the blocks with:

```js
for (const [cidStr, bytes] of this.store.entries()) {
    yield { cid: CID.parse(cidStr), bytes }
}
```

But, as we target `es5` we cannot use this with the default configuration given that the Map iteration was [added in `ES2015`](https://mariusschulz.com/blog/downlevel-iteration-for-es3-es5-in-typescript#using-downlevel-iteration-with-es2015-collections). 

As a result, we get the following error without using this:

```
src/blockstore/memory.ts:14:35 - error TS2569: Type 'IterableIterator<[string, Uint8Array]>' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.

14     for (const [cidStr, bytes] of this.store.entries()) {
```

Reply:
We could target something more recent than `es5`. Let's match `@ipld/car` and target ES2018 https://github.com/ipld/js-car/blob/e7d930087dd3e66bba425402c29c0d1e780f20c4/tsconfig.json#L19

Reply:
good call, done